DirectProgramming/DPC++FPGA/ReferenceDesigns/mvdr_beamforming/src/InputDemux.hpp

```diff
@@ -0,0 +1,264 @@
+#ifndef __INPUT_DEMUX_HPP__
+#define __INPUT_DEMUX_HPP__
+
+#include <CL/sycl.hpp>
+#include <CL/sycl/INTEL/fpga_extensions.hpp>
+#include <cmath>
+
+// utility classes
+#include "Tuple.hpp"
+#include "UnrolledLoop.hpp"
+
+#include "mvdr_complex.hpp"
+
+using namespace sycl;
+
+// SubmitInputDemuxKernel
+// Accept data from an input pipe, and split the data into two output pipes
+// based on header packets detected in the data stream.  Buffer up a full
+// training array and matching data samples before passing any data downstream,
+// so that the correct amount of each type of data is guaranteed to all
+// downstream kernels, and no partial matrices are ever transferred.
+template <typename InputDemuxKernelName,  // Name to use for the Kernel
+
+          size_t k_pipe_width,         // Number of complex numbers transferred
```

Comment:
Why the "k_" prefix?    Does this mean "kernel" in this case?

Reply:
As noted above, k_ indicates a compile time constant.